### ISSUE TITLE
Add Doc Comment For `RegistryMetadata._init_object`

### ISSUE THREAD

**Issue by biniona** [duolingo/minject#54]
Currently, `RegistryMetadata._init_object` has no explanation of what it is trying to do. In broad strokes, method recursively resolves the bindings `Deferred` bindings attached to the object being initialized, however there may be some subtlety here that I missed or forgot about.

This issue captures the task of determining exactly what this method does, documenting what it does, and documenting any relevant differences between `_init_object` and `_ainit_object`.

This Issue is dependent on #49.
